### PR TITLE
Attempt to turn on some more remote tests

### PR DESCRIPTION
**Pull request by rhatdan**
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@containers/podman-maintainers PTAL
This one looks like it is finally able to go in.
@edsantiago Turns on a bunch of tests to make you happy. :^)
Also explains why some Remote tests have to be skipped.
Finally diagnosed a few more podman-remote failures, that still need to be worked on.

File: test/e2e/namespace_test.go [hunk 1]
```diff
@@ -33,7 +33,14 @@ var _ = Describe("Podman namespaces", func() {
...
 	It("podman namespace test", func() {
-		SkipIfRemote("FIXME This should work on Remote")
+		if IsRemote() {
+			// --namespace flag not supported in podman remote
+			podman1 := podmanTest.Podman([]string{"--namespace", "test1", "run", "-d", ALPINE, "echo", "hello"})
+			podman1.WaitWithDefaultTimeout()
+			Expect(podman1.ExitCode()).To(Equal(125))
+			Expect(podman1.ErrorToString()).To(ContainSubstring("unknown flag: --namespace"))
+			return
+		}
```
Comment (edsantiago) [inline test/e2e/namespace_test.go:43]: Silly question, but why not move the `if IsRemote()` down a couple of lines, and avoid duplicating the command & wait?
```go
    podman1.WaitWithDefaultTimeout()
    if IsRemote() {
        Expect(125) // and message
    } else {
        Expect(0)  // and message
    }
```

File: test/e2e/run_env_test.go [hunk 2]
```diff
@@ -90,7 +90,15 @@ var _ = Describe("Podman run", func() {
 	})
 
 	It("podman run --env-host environment test", func() {
-		SkipIfRemote("FIXME, We should check that --env-host reports correct error on podman-remote")
+		if IsRemote() {
+			env := append(os.Environ(), "FOO=BAR")
+			session := podmanTest.PodmanAsUser([]string{"run", "--rm", "--env-host", ALPINE, "/bin/printenv", "FOO"}, 0, 0, "", env)
```
Comment (edsantiago) [inline test/e2e/run_env_test.go:95]: Similar to above, can we eliminate some duplication?

File: test/e2e/namespace_test.go (hunk 1 above)
Comment (rhatdan) [reply test/e2e/namespace_test.go:43]: Fixed

File: test/e2e/run_env_test.go (hunk 2 above)
Comment (rhatdan) [reply test/e2e/run_env_test.go]: Fixed

File: test/e2e/search_test.go [hunk 3]
```diff
@@ -237,7 +237,7 @@ registries = ['{{.Host}}:{{.Port}}']`
 	})
 
 	It("podman search attempts HTTP if registry is in registries.insecure and force secure is false", func() {
-		SkipIfRemote("FIXME This should work on podman-remote")
+		SkipIfRemote("--tls-verify is not supportedon podman-remote search")
```
Comment (edsantiago) [inline test/e2e/search_test.go:240]: [not worth re-pushing; please fix only if something else merits it] : "supportedon" needs space. Also lines 281, 320 below.

File: test/system/070-build.bats [hunk 4]
```diff
@@ -174,12 +174,19 @@ EOF
     run_podman build -t build_test -f build-test/Containerfile build-test
     local iid="${lines[-1]}"
 
+
+    if is_remote; then
+        ENVHOST=""
+    else
+	ENVHOST="--env-host"
```
Comment (edsantiago) [inline test/system/070-build.bats:181]: [another not-worth-pushing-just-for-this]: inconsistent indentation

File: test/e2e/system_df_test.go [hunk 5]
```diff
@@ -48,14 +48,19 @@ var _ = Describe("podman system df", func() {
 		session.WaitWithDefaultTimeout()
 		Expect(session.ExitCode()).To(Equal(0))
 
+		session = podmanTest.Podman([]string{"images", "-q"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+		totImages := strconv.Itoa(len(session.OutputToStringArray()))
```
Comment (edsantiago) [inline test/e2e/system_df_test.go:54]: Thank you! That has bitten us before.

Purely out of curiosity, is it ever possible for this number to be different from `podman info --format '{{.Store.ImageStore.Number}}'`?

**Comment by edsantiago** [comment]
This must've been incredibly tedious, iterating over and over so many tests. It is much appreciated. LGTM.

**Pull request merged**